tasks/rule_generator/src/main.rs

```diff
@@ -35,25 +36,39 @@ struct Context<'a> {
     fail_cases: &'a str,
 }
 
-fn parse_test_code<'a>(expr: &'a Expression) -> Option<Cow<'a, str>> {
-    match expr {
-        Expression::StringLiteral(lit) => Some(Cow::Borrowed(lit.value.as_str())),
-        Expression::TemplateLiteral(lit) => Some(Cow::Borrowed(lit.quasi().unwrap().as_str())),
+fn parse_test_code<'a>(source_text: &'a str, expr: &'a Expression) -> Option<Cow<'a, str>> {
+    let (test_code, option_code) = match expr {
+        Expression::StringLiteral(lit) => (Some(Cow::Borrowed(lit.value.as_str())), None),
+        Expression::TemplateLiteral(lit) => {
+            (Some(Cow::Borrowed(lit.quasi().unwrap().as_str())), None)
+        }
         Expression::ObjectExpression(obj_expr) => {
+            let mut test_code = None;
+            let mut option_code: Option<Cow<'_, str>> = None;
             for obj_prop in &obj_expr.properties {
                 match obj_prop {
                     ObjectProperty::Property(prop) => match &prop.key {
                         PropertyKey::Identifier(ident) if ident.name == "code" => match &prop.value
                         {
-                            PropertyValue::Expression(expr) => return parse_test_code(expr),
+                            PropertyValue::Expression(expr) => {
+                                let Expression::StringLiteral(s) = expr else {
+                                return None;
+                              };
```

Comment:
Is it guaranteed that the value of the `code` key will always be a string across all rule tests? I haven't checked myself

Reply:
I've looked at tens of files or so and it is all strings.